### PR TITLE
Janitoring: address some portability issues

### PR DESCRIPTION
**Pull request by akva2**
Some small fixes to improve portability and compiler compatibility.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp [hunk 1]
```diff
@@ -1549,7 +1549,7 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
             std::vector<double> volume(this->m_nactive);
 
             #pragma omp parallel for schedule(static)
-            for (std::size_t active_index = 0; active_index < this->m_active_to_global.size(); active_index++) {
+            for (std::int64_t active_index = 0; active_index < static_cast<std::int64_t>(this->m_active_to_global.size()); active_index++) {
```
Comment (atgeirr) [inline opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1552]: This made me a little curious, what is the reason for this change? Is it due to OpenMP issues?

**Review by atgeirr — APPROVED** [review]

File: opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp (hunk 1 above)
Comment (akva2) [reply opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1552]: I tried to explain in the commit message.

Comment (atgeirr) [reply opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp:1552]: Did not notice the commit message, sorry for the noise!
